### ISSUE TITLE
Implement `lowerBatchNorm` UOp Lowering Function

### ISSUE THREAD

**Issue by Hitomamacs** [ZantFoundation/Z-Ant#358]
# Feature Request: Implement `lowerBatchNorm` UOp Lowering Function

## Summary
Implement the `lowerBatchNorm` function to lower the ONNX BatchNormalization operation into UOps for execution in the Z-Ant tensor framework.

## Description
The Z-Ant codebase currently has:
- ✅ Tensor math implementation: `op_batch_norm.zig` 
- ✅ IR graph operator: `op_batch_norm.zig` in IR_graph
- ❌ **Missing UOps lowering function: `lowerBatchNorm`**

## ONNX Specification
**ONNX BatchNormalization Documentation**: https://onnx.ai/onnx/operators/onnx__BatchNormalization.html

**Operation**: BatchNormalization
- **Domain**: `""`  
- **Since version**: 15
- **Function**: Carries out batch normalization as described in the paper https://arxiv.org/abs/1502.03167. Depending on the mode it is being run, there are five required and one optional parameter
- **Attributes**: 
  - `epsilon` (float, default=1e-05): The epsilon value to use to avoid division by zero
  - `momentum` (float, default=0.9): Factor used in computing the running mean and variance (training mode only)
  - `training_mode` (int, default=0): If set to 1, use the batch statistics. If set to 0, use the running statistics
- **Inputs**: 
  - `X` (T): Input tensor
  - `scale` (T1): Scale tensor of shape (C,)
  - `B` (T1): Bias tensor of shape (C,)
  - `input_mean` (T2): Running mean tensor of shape (C,)
  - `input_var` (T2): Running variance tensor of shape (C,)
- **Outputs**:
  - `Y` (T): Output tensor
  - `output_mean` (T2, optional): Running mean after update
  - `output_var` (T2, optional): Running variance after update

### Function Signature
```zig
pub fn lowerBatchNorm(
    b: *UOpBuilder,
    input_id: usize,
    scale_id: usize,
    bias_id: usize,
    mean_id: usize,
    var_id: usize,
    out_shape: []const usize,
    out_dtype: DType,
    epsilon: f32,
    training_mode: bool,
) usize
```

### Implementation Approach
Batch normalization requires channel-wise statistics computation:

1. **Setup Phase**:
   - Create VIEWs for input, scale, bias, mean, variance tensors
   - Define global output buffer with `DEFINE_GLOBAL`
   - Calculate channel dimension and spatial dimensions

2. **Training Mode Algorithm**:
   
   **Pass 1 - Compute Batch Statistics** (if training_mode=true):
   - `RANGE` over batch and spatial dimensions
   - Calculate mean per channel: μ = (1/N) * Σ(x_i)
   - Calculate variance per channel: σ² = (1/N) * Σ((x_i - μ)²)
   
   **Pass 2 - Normalize**:
   - `RANGE` over all elements
   - Load input value, scale, bias, mean, variance for current channel
   - Apply normalization: y = scale * (x - mean) / sqrt(variance + epsilon) + bias
   - `STORE` result

3. **Inference Mode Algorithm**:
   - Single pass using provided running statistics
   - Direct normalization with precomputed mean/variance

4. **Return**: SSA id of output buffer

### Expected UOp Sequence (Training Mode)
```
VIEW(input,scale,bias,mean,var) -> DEFINE_GLOBAL(output) -> 
RANGE(compute_stats) -> LOAD -> ADD(sum) -> ENDRANGE ->
DIV(count) -> // compute mean
RANGE(compute_var) -> LOAD -> SUB(mean) -> MUL -> ADD(var_sum) -> ENDRANGE ->
DIV(count) -> // compute variance
RANGE(normalize) -> LOAD -> SUB(mean) -> DIV(sqrt(var+eps)) -> MUL(scale) -> ADD(bias) -> STORE -> ENDRANGE
```

## Mathematical Definition
For training mode:
1. μ_c = (1/N*H*W) * Σ(x_{n,c,h,w}) over n,h,w
2. σ²_c = (1/N*H*W) * Σ((x_{n,c,h,w} - μ_c)²) over n,h,w
3. y_{n,c,h,w} = γ_c * (x_{n,c,h,w} - μ_c) / √(σ²_c + ε) + β_c

Where:
- γ (scale), β (bias) are learnable parameters
- ε (epsilon) prevents division by zero
- C is number of channels, N is batch size

## Files to Modify
- **Primary**: Create the `lowerBatchNorm` function in the appropriate UOps lowering module
- **Integration**: Add batch normalization support to UOp processing pipeline
- **Testing**: Verify compatibility with existing `op_batch_norm.zig` tensor math implementation

## Acceptance Criteria
- [ ] `lowerBatchNorm` function implemented following Z-Ant UOp patterns
- [ ] Function correctly handles inference modes
- [ ] Integration with existing tensor math and IR graph implementations
- [ ] Proper memory management and SSA id handling
- [ ] Compatible with Z-Ant's broadcasting and shape handling systems
- [ ] Numerical stability with epsilon parameter
- [ ] Support for different tensor layouts (NCHW primarily)

## Related Files
- `/src/Core/Tensor/TensorMath/lib_elementWise_math/op_batch_norm.zig` (tensor math)
- `/src/IR_graph/op_union/operators/op_batch_norm.zig` (IR graph)  
- `/src/Core/Tensor/TensorMath/lib_elementWise_math/op_addition.zig` (reference implementation)

## Priority
High - Batch normalization is essential for training stability and convergence in deep neural networks.

## Implementation Notes
- Complex multi-pass algorithm for training mode
- Requires reduction operations across spatial dimensions
- Memory access patterns should consider channel locality
- Training mode requires updating running statistics (optional outputs)
- Consider numerical precision for variance computation
